bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/internal/validation/ConfigDescriptionValidatorImpl.java

```diff
@@ -101,8 +102,9 @@ public void validate(Map<String, Object> configurationParameters, URI configDesc
 
         Collection<ConfigValidationMessage> configDescriptionValidationMessages = new ArrayList<>();
 
-        for (String key : map.keySet()) {
-            ConfigDescriptionParameter configDescriptionParameter = map.get(key);
+        for (Entry<String, ConfigDescriptionParameter> entry : map.entrySet()) {
+            String key = entry.getKey();
+            ConfigDescriptionParameter configDescriptionParameter = entry.getValue();
             if (configDescriptionParameter != null) {
```

Comment:
I think this null-check can be omitted if we use entry.getValue() because the map has no null values.